workloads/uber/src/main/java/io/vectorized/tx_subscribe/Workload.java

```diff
@@ -324,6 +324,53 @@ private void producingProcess(int pid, int partition) throws Exception {
                 log(pid, "send\t" + op.oid);
                 producer.beginTransaction();
                 offset = producer.send(new ProducerRecord<String, String>(args.source, partition, args.server, "" + op.oid)).get().offset();
```

Comment:
I wonder if we build a client library state machine to make Kafka transactions easier to program against from a client side and open source it

Reply:
It feels like it's an uphill battle, we may write a blog post / article pointing to a repo full with examples